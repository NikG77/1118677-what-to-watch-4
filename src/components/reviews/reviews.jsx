import React from "react";
// import PropTypes from "prop-types";
// import {filmType} from "../../types";
import {reviews} from "../../mock/films.js";

const options = {year: `numeric`, month: `long`, day: `numeric`};

const reformatDate = (date) => new Intl.DateTimeFormat(`en-EN`, options).format(new Date(date));
const getForDateTime = (date) => (`${date.getFullYear()}-${date.getMonth() + 1}-${date.getDay()}`);

const Reviews = () => {

  const reviewsCol1 = reviews.filter((review, i) => !(i % 2));

  const reviewsCol2 = reviews.filter((review, i) => i % 2);

  return (
    <React.Fragment>
      <div className="movie-card__reviews movie-card__row">
        <div className="movie-card__reviews-col">

          {reviewsCol1.map((review) => (
            <div key={review.id} className="review">
              <blockquote className="review__quote">
                <p className="review__text">{review.text}</p>

                <footer className="review__details">
                  <cite className="review__author">{review.author}</cite>
                  <time className="review__date" dateTime={getForDateTime(review.date)}>{reformatDate(review.date)}</time>
                </footer>
              </blockquote>

              <div className="review__rating">{review.rating}</div>
            </div>

          ))}

        </div>
        <div className="movie-card__reviews-col">
          {reviewsCol2.map((review) => (
            <div key={review.id} className="review">
              <blockquote className="review__quote">
                <p className="review__text">{review.text}</p>

                <footer className="review__details">
                  <cite className="review__author">{review.author}</cite>
                  <time className="review__date" dateTime={getForDateTime(review.date)}>{reformatDate(review.date)}</time>
                </footer>
              </blockquote>

              <div className="review__rating">{review.rating}</div>
            </div>
          ))}

        </div>
      </div>
    </React.Fragment>
  );

};

// Reviews.propTypes = {
//   rewiews: PropTypes.arrayOf(PropTypes.share({
//     text: PropTypes.string.isRequired,
//     author: PropTypes.string.isRequired,
//     date: PropTypes.date.isRequired,
//     rating: PropTypes.number.isRequired,

//   })),
// };

export default Reviews;
