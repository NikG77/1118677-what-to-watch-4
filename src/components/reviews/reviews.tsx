import * as React from "react";
import {connect} from "react-redux";

import {Comment} from "../../types";
import ReviewsColumn from "../reviews-column/reviews-column";
import {getComments} from "../../reducer/data/selectors";

interface Props {
  id: number;
  reviews: Array<Comment>;
}

const Reviews: React.FunctionComponent<Props> = (props: Props) => {

  const {reviews} = props;
  if (reviews) {
    const reviewsColumns1 = reviews.slice(0, Math.ceil(reviews.length / 2));
    const reviewsColumns2 = reviews.slice(Math.ceil(reviews.length / 2));

    return (
      <div className="movie-card__reviews movie-card__row">
        {reviewsColumns1.length > 0 && <ReviewsColumn reviewsColumns={reviewsColumns1} />}
        {reviewsColumns2.length > 0 && <ReviewsColumn reviewsColumns={reviewsColumns2} />}
      </div>
    );
  }
  return null;
};


const mapStateToProps = (state) => ({
  reviews: getComments(state),
});

export {Reviews};

export default connect(mapStateToProps)(Reviews);

