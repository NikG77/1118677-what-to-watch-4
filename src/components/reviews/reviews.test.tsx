import * as React from "react";
import * as renderer from "react-test-renderer";
import {Reviews} from "./reviews";


it(`Should Reviews render correctly`, () => {
  const tree = renderer
    .create(
        <Reviews
          id={6}
          reviews={[]}

        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
