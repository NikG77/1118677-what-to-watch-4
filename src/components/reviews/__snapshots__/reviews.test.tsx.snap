// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Reviews render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
/>
`;
